libcaf_net/src/udp_datagram_socket.cpp

```diff
@@ -100,9 +100,8 @@ variant<std::pair<size_t, ip_endpoint>, sec> read(udp_datagram_socket x,
                          << CAF_ARG(buf.size()) << " of " << CAF_ARG(num_bytes)
                          << " bytes");
     ip_endpoint ep;
-    // TODO: how to properly pass error further?
     if (auto err = detail::convert(addr, ep))
-      return sec::runtime_error;
```

Comment:
I think we should add `CAF_ASSERT(err.category == atom("system"))` here to make sure the `static_cast` below is safe.